src/main/java/bio/terra/service/filedata/azure/tables/TableDirectoryDao.java

```diff
@@ -176,17 +176,27 @@ public FireStoreDirectoryEntry retrieveByPath(
         .orElse(null);
   }
 
-  public List<UUID> validateRefIds(TableServiceClient tableServiceClient, List<UUID> refIdArray) {
+  public List<String> validateRefIds(
+      TableServiceClient tableServiceClient, List<String> refIdArray) {
     logger.info("validateRefIds for {} file ids", refIdArray.size());
-    List<UUID> missingIds = new ArrayList<>();
-    for (UUID s : refIdArray) {
-      ListEntitiesOptions options =
-          new ListEntitiesOptions().setFilter(String.format("fileId eq '%s'", s.toString()));
-      if (!TableServiceClientUtils.tableHasEntries(tableServiceClient, TABLE_NAME, options)) {
-        missingIds.add(s);
-      }
-    }
-    return missingIds;
+    TableClient tableClient = tableServiceClient.getTableClient(TABLE_NAME);
+    return Lists.partition(refIdArray, MAX_FILTER_CLAUSES).stream()
+        .flatMap(
+            refIds -> {
+              var filter =
+                  refIds.stream()
+                      .map(refId -> String.format("fileId eq '%s'", refId))
+                      .collect(Collectors.joining(" or "));
```

Comment:
Is there no `in` operator?

Reply:
Unfortunately, no https://docs.microsoft.com/en-us/rest/api/storageservices/querying-tables-and-entities#supported-comparison-operators

Reply:
Yeah...sadness

Reply:
That's weird, I was able to find an example that looks like it did work. https://docs.microsoft.com/en-us/odata/concepts/queryoptions-overview:

> - All products whose name value is ‘Milk’ or ‘Cheese’: http://host/service/Products?$filter=Name in ('Milk', 'Cheese')

Reply:
Worth trying but that's OData doc, not azure tables.  I'm not sure azure tables has all of the Odata spec implemented

Reply:
This does seem to work, not sure where it is in the docs. You can test it out using the OData reference service. https://services.odata.org/TripPinRESTierService/People?$filter=FirstName%20in%20(%27Scott%27)

Reply:
Here's a doc link that mentions `in` http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_BuiltinFilterOperations

Reply:
Hmm...let me test it out.  Again, just because OData supports it doesn't mean azure tables does

Reply:
Yeah, I don't think it works unfortunately:

![image](https://user-images.githubusercontent.com/5633787/132713427-e0d390ad-4d03-49f9-903c-f490d29a1488.png)

Reply:
😞